### PR TITLE
Fix bug ConcurrentModificationException when eraseTables

### LINKED ISSUE #4196: ConcurrentModificationException when eraseTables
### Steps to reproduce the behavior (Required)
In some situation.

### Real behavior (Required)
2022-03-17 15:03:59,704 ERROR (recycle bin|25) [Daemon.run():121] daemon thread got exception. name: recycle bin
java.util.ConcurrentModificationException: null
        at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:719) ~[?:1.8.0_321]
        at java.util.LinkedHashMap$LinkedKeyIterator.next(LinkedHashMap.java:742) ~[?:1.8.0_321]
        at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:47) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:47) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.catalog.CatalogRecycleBin.eraseTable(CatalogRecycleBin.java:265) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.CatalogRecycleBin.runAfterCatalogReady(CatalogRecycleBin.java:676) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:61) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:119) [starrocks-fe.jar:?]
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.3

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4196

## Problem Summary(Required) ：

This issue cause by tableIter.remove(). Take out the logic of deletion to solve this problem.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|16|16|100.00%|

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Review by gengjun-git — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 17 / 17 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|17|17|100.00%|

**Review by stdpain — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java [hunk 1]
```diff
@@ -276,24 +276,28 @@ private synchronized void eraseTable(long currentTimeMs) {
...
+                    tableToRemove.add(tableInfo);
                     currentEraseOpCnt++;
                     if (currentEraseOpCnt >= MAX_ERASE_OPERATIONS_PER_CYCLE) {
                         break;
                     }
                 }
             } // end for tables
-            if (!tableIdList.isEmpty()) {
-                Catalog.getCurrentCatalog().getEditLog().logEraseMultiTables(tableIdList);
-                LOG.info("multi erase write log finished. erased {} table(s)", currentEraseOpCnt);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java:295]: Better to check `currentEraseOpCnt` in the outer loop.

**Pull request merged**